### ISSUE TITLE
Dynamic web server content is being cached by CDN

### ISSUE THREAD

**Issue by lukehutch** [serverpod/serverpod#2534]
When I deploy a new version of my server to GCP, the web page content dynamically served by Relic does not change. I logged into the VM instance, created a shell in the Docker container, and verified that the new content is there, but it is not showing in my browser. I have waited over 12 hours for the changes to show, and they have not. Therefore, I assume that these dynamic pages are being cached by a CDN with infinite (or large) TTL.

I am serving only static content, but I wrapped every static page in a `Route` so that it would be served dynamically, because of [this note](https://docs.serverpod.dev/concepts/webserver#special-widgets-and-routes) in the docs:

> Static files are configured to be cached hard by the web browser and through Cloudfront's content delivery network (if you use the AWS deployment). If you change static files, they will need to be renamed, or users will most likely access old files. To make this easier, you can add a version number when referencing the static files. The version number will be ignored when looking up the actual file. E.g., /static/my_image@v42.png will serve to the /static/my_image.png file. More advanced cache management will be coming to a future version of Serverpod.

This doesn't say what the CDN policy is on GCP, but I would not expect dynamic pages to be cached by a CDN.

**Comment by lukehutch** [comment]
Here is the CDN policy:

![image](https://github.com/user-attachments/assets/eec00977-bdd0-4238-9f8c-d7f96f0b7967)

How does the CDN determine what is static and what is dynamic content?

**Comment by vlidholt** [comment]
This is most likely depending on your headers on the files you are requesting.

**Comment by lukehutch** [comment]
Yes, that's exactly my point -- Relic needs to return no-cache headers for dynamic routes.

**Comment by lukehutch** [comment]
#2539 allows static content whose path matches a regexp to not be cached. However, the same header change needs to be made for all dynamic content.

**Comment by SandPod** [comment]
Dynamic routes should not be cached, when quickly inspecting the code it does not look like we set any headers for caching at all.

We might need to add a default no-cache header for the dynamic routes.
